src/test/java/spleetwaise/transaction/model/transaction/DateTest.java

```diff
@@ -31,6 +31,8 @@ public void isValidDate_null_exceptionThrown() {
     @Test
     public void isValidDate_invalidInput_returnsFalse() {
         assertFalse(Date.isValidDate(""));
+        assertFalse(Date.isValidDate("     "));
+        assertFalse(Date.isValidDate("30022024")); // Feb edge case
```

Comment:
Recommend to refer to week 11 tutorial recording to add test cases based on the date equivalence partion: [1-28],[29],[30],[31]. Some years's feb accept 29 as the day, some months dont accept 31.

Reply:
In this case, the validation might be harder since we are validating by the whole date rather than just the day fields.

Reply:
I see, could u point us to the documentation?

Reply:
You mean the current implementation still allows invalid date? Or the test cases needs more work